src/test/java/tests/pages/DashboardPage.java

```diff
@@ -0,0 +1,121 @@
+
+import java.util.List;
+
+import static core.utils.Timeout.TWO_SEC;
+import static org.junit.Assert.assertTrue;
+
+public class DashboardPage extends PageFactory {
+  public static final String TITLE_ALL_DASHBOARDS_XPATH = "//span[@title = 'All Dashboards']";
+  public static final String DEMO_DASHBOARD_LINE_XPATH = "//div[contains(@class, 'gridRow')]//a[text() = 'DEMO DASHBOARD']";
+  public static final String WIDGETS_GRID_XPATH = "//div[contains(@class, 'react-grid-item widgetsGrid')]";
+  public static final String RESIZE_WIDGETS_BUTTON_XPATH = "//span[@class ='react-resizable-handle react-resizable-handle-se']";
+  public static final String WIDGET_XPATH_TEMPLATE = "//div[contains(@class, 'react-grid-item widgetsGrid') and contains(., '%s')]";
+  private final WebDriver driver;
+
+  public DashboardPage(WebDriver driver) {
+    this.driver = driver;
+    initElements(driver, this);
+  }
+
+  public void waitTillPageLoaded() {
+    elementVisibility(By.xpath(TITLE_ALL_DASHBOARDS_XPATH));
+  }
+
+  public void navigateToDashboardWidgets() {
+    WebElement element = driver.findElement(By.xpath(DEMO_DASHBOARD_LINE_XPATH));
+    clickOnElementUsingJs(element);
+
+    elementVisibility(By.xpath(WIDGETS_GRID_XPATH));
+  }
+
+  public void scrollToTheWidget(final String widget) {
+
+    Actions action = new Actions(driver);
+    boolean state = isWidgetElementIsVisible(widget);
+    while (state) {
+      //change focus and star scrolling
+      driver.findElement(By.xpath(WIDGETS_GRID_XPATH)).click();
+      action.sendKeys(Keys.PAGE_DOWN).build().perform();
+      state = isWidgetElementIsVisible(widget);
+    }
+
+    WebElement searchedWidget = driver.findElement(By.xpath(WIDGET_XPATH_TEMPLATE.formatted(widget)));
+    ((JavascriptExecutor) driver).executeScript("arguments[0].scrollIntoView(true);", searchedWidget);
+  }
+
+  public void isWidgetVisible(final String widget) {
+    elementVisibility(By.xpath(WIDGET_XPATH_TEMPLATE.formatted(widget)));
+  }
+
+  public void navigateToTheWidgetVisible(final String widget) {
+    clickOnElementUsingJs(driver.findElement(By.xpath(WIDGET_XPATH_TEMPLATE.formatted(widget))));
+  }
+
+  private boolean isWidgetElementIsVisible(final String widgetName) {
+    List<WebElement> widgets = driver.findElements(By.xpath(WIDGETS_GRID_XPATH));
+    return widgets.stream().noneMatch(el -> el.getText().contains(widgetName));
+  }
+
+  public void resizeWidget(final String widgetName) {
+    String widget = WIDGET_XPATH_TEMPLATE.formatted(widgetName);
+
+    WebElement widgetToResize = driver.findElement(By.xpath(widget));
+    Dimension oldElementSize = widgetToResize.getSize();
+
+    WebElement scrollOfElement = driver.findElement(By.xpath(widget + RESIZE_WIDGETS_BUTTON_XPATH));
+
+    resizeElement(scrollOfElement, true);
+    Dimension newElementSize = widgetToResize.getSize();
+
+    verifyDimensionChanges(oldElementSize, newElementSize);
+    resizeElement(scrollOfElement, false);
```

Comment:
if your verification method fails your resize for returning the widget size isn't perform so you should think about another way of performing resizeElement a second time. It can be either using soft assertion or extracting the method to the @After method